src/plugins/workspace/server/workspace_client.ts

```diff
@@ -167,7 +167,7 @@ export class WorkspaceClientWithSavedObject implements IWorkspaceDBImpl {
         }
       }
       await client.update<Omit<WorkspaceAttribute, 'id'>>(WORKSPACE_TYPE, id, attributes, {
-        permissions,
+        permissions: convertToFullPermissions(permissions),
```

Comment:
I don't understand the root cause of the issue, could you elaborate a bit why adding empty permissions?

Reply:
We don't know the original permissions for one object, we need to generate a new permissions with all permission types. It will clear not used principals in the saved objects.

Reply:
the doc in parameter is partial doc, guess there has merge logic in `client.update` method https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/7.17/api-reference.html#_update

Reply:
@Hailong-am what's the behavior of `POST /<index>/_update/<_id>`? will it create the object if the id does not exist?

Reply:
@ruanyl `The update API also supports passing a partial document, which is merged into the existing document`. if the id is not exists, 404 will return.